### PR TITLE
feat: stale filters cleaner

### PR DESCRIPTION
**Pull request by allnil**
#5038
Hey!

Stale filters cleaner
Maybe should add a unit test on this thing

### CODE REVIEW

File: crates/rpc/rpc/src/eth/filter.rs [hunk 1]
```diff
@@ -52,8 +59,25 @@ impl<Provider, Pool> EthFilter<Provider, Pool> {
...
+        inner.task_spawner.spawn_critical(
+            "stale-filters-cleaner",
+            Box::pin(async move {
+                let mut interval = tokio::time::interval(stale_filter_ttl);
+                loop {
+                    interval.tick().await;
+                    {
+                        active_filters_arc.lock().await.retain(|_, filter| {
+                            filter.last_poll_timestamp.elapsed() < stale_filter_ttl
+                        });
```
Comment (mattsse) [inline crates/rpc/rpc/src/eth/filter.rs:76]: I think we want some traces here:

1. after interval triggered 
2. trace the filter id that we evict

File: crates/rpc/rpc/src/eth/filter.rs [hunk 2]
```diff
@@ -43,6 +49,7 @@ impl<Provider, Pool> EthFilter<Provider, Pool> {
         eth_cache: EthStateCache,
         max_logs_per_response: usize,
         task_spawner: Box<dyn TaskSpawner>,
+        stale_filter_ttl: Duration,
```
Comment (mattsse) [inline crates/rpc/rpc/src/eth/filter.rs:52]: we should add a note to the docs that this spawns a service task

File: crates/rpc/rpc/src/eth/filter.rs [hunk 3]
```diff
@@ -52,8 +59,25 @@ impl<Provider, Pool> EthFilter<Provider, Pool> {
...
+
+        inner.task_spawner.spawn_critical(
+            "stale-filters-cleaner",
+            Box::pin(async move {
+                let mut interval = tokio::time::interval(stale_filter_ttl);
+                loop {
+                    interval.tick().await;
+                    {
+                        active_filters_arc.lock().await.retain(|_, filter| {
+                            filter.last_poll_timestamp.elapsed() < stale_filter_ttl
```
Comment (mattsse) [inline crates/rpc/rpc/src/eth/filter.rs:75]: let's make this an async function of `EthFilter` instead, and move the `stale_filter_ttl` to a field,

so the service task only needs to call a function

**Review by mattsse — CHANGES_REQUESTED** [review]
this is great,

only have pedantic style nits

File: crates/rpc/rpc/src/eth/filter.rs (hunk 2 above)
Comment (allnil) [reply crates/rpc/rpc/src/eth/filter.rs:52]: Good! 
I added a comment in the struct definition, should I add something else?

File: crates/rpc/rpc/src/eth/filter.rs (hunk 3 above)
Comment (allnil) [reply crates/rpc/rpc/src/eth/filter.rs]: Ok! Done!

**Review by mattsse — APPROVED** [review]
this is great,

I made a few small cosmetic changes and an optimization for the elapsed check, it's fine to use a single `Instant::now` for all entries

**Comment by allnil** [comment]
Cool!
Thank you @mattsse for your comments!

**Pull request merged**